cmake/CMakeLists.txt

```diff
@@ -904,14 +930,16 @@ if (date_FOUND)
   message("Use date from preinstalled system lib")
   add_library(date_interface ALIAS date::date)
 else()
-  message("Use date from submodule")
-  add_subdirectory(external/date EXCLUDE_FROM_ALL)
+    FetchContent_Declare(
+		  date
+		  URL ${DEP_URL_date}
+		  URL_HASH SHA1=${DEP_SHA1_date}
+		)
```

Comment:
I am wondering if we could move all these Fetches into a separate included file? It kinda distracts from the main logic of the main makefile.

Reply:
Then it may have logic like:
```
if ( (onnxruntime_ENABLE_TRAINING OR onnxruntime_ENABLE_TRAINING_OPS) AND onnxruntime_ENABLE_TRAINING_TORCH_INTEROP)
   FetchContent(...)
endif()
```
Is it ok？

Reply:
It would be Okay with me, because it clearly shows what and when needs that content.